src/core/server/uuid/resolve_uuid.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import uuid from 'uuid';
+import { promisify } from 'util';
+import * as fs from 'fs';
+import { join } from 'path';
+import { take } from 'rxjs/operators';
+import { IConfigService } from '../config';
+import { PathConfigType, config as pathConfigDef } from '../path';
+import { HttpConfigType, config as httpConfigDef } from '../http';
+import { Logger } from '../logging';
+
+const FILE_ENCODING = 'utf8';
+const FILE_NAME = 'uuid';
+
+const readFile = promisify(fs.readFile);
+const writeFile = promisify(fs.writeFile);
```

Comment:
nit: Node v10 introduced promisified versions of fs api's https://nodejs.org/docs/latest-v10.x/api/fs.html#fs_fspromises_readfile_path_options

Reply:
TIL